### PR TITLE
Run make lint on all operating systems

### LINKED ISSUE #3489: make lint does not properly catch CI failures when committing changes into master
For some reason we can have a green master, but when I make changes off of that I suddenly get lots of "make lint" errors that I have to then fix in the next PR, which makes those PRs harder to review. We need to debug why this happens and fix it.

### PR DESCRIPTION
**Pull request by fschutt**
Fixes #3489.

Continuation of https://github.com/wasmerio/wasmer/pull/3491.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -34,26 +34,78 @@ jobs:
...
   lint:
-    name: Code lint
-    runs-on: ubuntu-22.04
+    name: Code lint on ${{ matrix.metadata.build }}
+    runs-on: ${{ matrix.metadata.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        metadata: [
+          {
```
Comment (theduke) [inline .github/workflows/test.yaml:43]: I'm not a big fan of adding 4 more extra jobs, we already have a lot of them.

Can we add the clippy tests to some other jobs that are relatively fast anyway?

Comment (fschutt) [reply .github/workflows/test.yaml]: Done in https://github.com/wasmerio/wasmer/pull/3499/commits/6ceb8ff312b3754c87d1d8405e0e28341dcfd0ce

**Review by theduke — APPROVED** [review]
LGTM

**Comment by fschutt** [comment]
@syrusakbary can be merged now, I don't think we need to spend more time on this

**Comment by theduke** [comment]
@fschutt conflicts!

File: Makefile [hunk 2]
```diff
@@ -849,9 +849,12 @@ update-testsuite:
 
 lint-packages: RUSTFLAGS += -D dead-code -D nonstandard-style -D unused-imports -D unused-mut -D unused-variables -D unused-unsafe -D unreachable-patterns -D bad-style -D improper-ctypes -D unused-allocation -D unused-comparisons -D while-true -D unconditional-recursion -D bare-trait-objects -D function_item_references # TODO: add `-D missing-docs`
 lint-packages:
-	RUSTFLAGS="${RUSTFLAGS}" cargo clippy --all --exclude wasmer-cli -- -D clippy::all
+	RUSTFLAGS="${RUSTFLAGS}" cargo clippy --all --exclude wasmer-cli --exclude wasmer-compiler-llvm $(compiler_features) -- -D clippy::all
```
Comment (theduke) [inline Makefile:852]: This should only be excluded if LLVM is disabled, not unconditionally.

**Review by theduke — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (fschutt) [reply Makefile]: Done in https://github.com/wasmerio/wasmer/pull/3499/commits/c83b9e0455d1a906ba105258e6760a177d2816d9

**Pull request closed without merging**